script.js

```diff
@@ -1,12 +1,130 @@
+
+let displayNum = document.createElement('div');  //how many episodes on the screen
+displayNum.id = "displayNum";
+   body.insertBefore(displayNum, rootElem);
+   displayNum.innerText = `displaying ${allEpisodes.length}/${allEpisodes.length} episodes.`;
+
+let dataFrom = document.createElement('p'); // element telling where data is from
+dataFrom.innerHTML = `This data is from <a href =${"https://www.tvmaze.com/shows/82/game-of-thrones"} >TVMaze.com</a> `;
+body.appendChild(dataFrom);
+
 
-function makePageForEpisodes(episodeList) {
-  const rootElem = document.getElementById("root");
-  rootElem.textContent = `Got ${episodeList.length} episode(s)`;
+let arr =[];   //Creating array of data needed
+
+ for(let i=0; i<numOfEp; i++){
+  // Zero pads to two digits for season and episode.
+     prepended_out = ("0" + allEpisodes[i].season ).slice(-2);
+   prepended_out2 = ("0" + allEpisodes[i].number ).slice(-2);
+let epData = {
+            name:`${allEpisodes[i].name}`,
+           season: `${prepended_out}`,
+           number: `${prepended_out2}`,
+           image: `${allEpisodes[i].image.medium}`,
+           summary: `${allEpisodes[i].summary}`,
+           id: `${allEpisodes[i].id}`
+          }; 
+   arr.push(epData);
 }
 
-window.onload = setup;
+
+//episode data to be displayed on screen
+function display(){
+  let data = arr.map((ep)=> {
+    prepended_out = ("0" + ep.season ).slice(-2);
+     prepended_out2 = ("0" + ep.number ).slice(-2);
+    return   `
+           <div class="episode" id="${ep.id}">
+               <h1>${ep.name} S${prepended_out}E${prepended_out2}</h1>
+                 <img src="${ep.image}" title ="ep"></img>
+                 ${ep.summary}
+              </div>`;
+    }) .join('');
+   return data;
+  }
+
+ rootElem.innerHTML = rootElem.innerHTML + display(); 
+
+  let episode = document.querySelector('#episode');
+//Search for episodes
+  search.addEventListener('keyup', (e) => {
+      const searchString = e.target.value.toLowerCase(); //value typed
+      const filteredEp = arr.filter((ep) => {
+          return (  
+            ep.name.toLowerCase().includes(searchString) || 
+           ep.summary.toLowerCase().includes(searchString)
+          );
+     
+      });  
+      
+        if(filteredEp.length > 0){
+          displayNum.innerText = `Displaying ${filteredEp.length}/${allEpisodes.length} episodes.`
+          //changes inner text depending on how many episodes displayed
+
+          rootElem.innerHTML =     //changes episodes displayed depending on input
+            filteredEp.map((ep) => { 
+                prepended_out = ("0" + ep.season ).slice(-2);
+                prepended_out2 = ( "0" + ep.number ).slice(-2);
+         return   `
+         <div class="episode" id="${ep.id}">
+             <h1>${ep.name} S${prepended_out}E${prepended_out2}</h1>
+             <img src="${ep.image}" title ="ep"></img>
+              ${ep.summary} </div> ` ;
+                        
+         }).join('');
```

Comment:
Great use of .joining, all the html into one string which u then assign to the inner html